aten/src/ATen/TensorImpl.h

```diff
@@ -17,11 +17,35 @@ struct Storage;
 struct Tensor;
 } // namespace at
 
+namespace at {
+struct TensorImplOptions {
+  bool has_storage_;
```

Comment:
can you make these const?

Reply:
It seems that I couldn't make them const because this line https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/variable.cpp#L164 changes `is_variable_` outside of the constructor. If I put everything into TensorImplOptions and make them const, I can't do either `options_.is_variable_ = true` or `options_ = TensorImplOptions(true)`, because `TensorImplOptions` wouldn't have a copy constructor that would make sense here.

Reply:
That line seems bogus to me; is_variable should always be true here.  Can you change that line to an AT_ASSERT?